homeassistant/components/unifiprotect/__init__.py

```diff
@@ -89,6 +93,34 @@ async def async_setup_entry(hass: HomeAssistant, entry: UFPConfigEntry) -> bool:
     bootstrap = protect.bootstrap
     nvr_info = bootstrap.nvr
     auth_user = bootstrap.users.get(bootstrap.auth_user_id)
+
+    # Check if API key is missing
+    if not protect.is_api_key_set() and auth_user and nvr_info.can_write(auth_user):
+        try:
+            new_api_key = await protect.create_api_key(
+                name=f"Home Assistant ({hass.config.location_name})"
+            )
+        except NotAuthorized as err:
+            _LOGGER.error("Failed to create API key: %s", err)
+        else:
+            protect.set_api_key(new_api_key)
+            hass.config_entries.async_update_entry(
+                entry, data={**entry.data, CONF_API_KEY: new_api_key}
+            )
+
+    if not protect.is_api_key_set():
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN,
+                context={"source": "reauth", "entry_id": entry.entry_id},
+                data=entry.data,
+            )
+        )
```

Comment:
They should get the popup already when you raise `ConfigEntryAuthFailed` so I don't think we need to start it manually as well.

```suggestion
```